resources/tentative/complex-static-html/src/big-dom-generator/gen-css.js

```diff
@@ -1,238 +1,201 @@
+    const validChildren = children.slice(0, currentIndex);
+    return random.choice(validChildren);
+};
+
 const getNextDepth = (combinator, depth) => {
     switch (combinator) {
-        case " ":
+        case Combinator.DESCENDANT:
             return random.randRange(0, depth);
-        case " > ":
+        case Combinator.CHILD:
             return depth - 1;
-        case " + ":
-        case " ~ ":
+        case Combinator.ADJACENT_SIBLING:
+        case Combinator.GENERAL_SIBLING:
             return depth;
         default:
             throw new Error(`Invalid combinator: ${combinator}`);
     }
 };
 
-const getNextIndex = (combinator, newDepth, index) => {
-    // prettier-ignore
-    if (combinator === " + ")
-        return index - 1;
-    // prettier-ignore
-    if (combinator === " ~ ")
-        return random.randRange(0, index);
-    switch (newDepth) {
-        case 6:
-            return index;
-        case 5:
-        case 4:
-            return 1;
-        case 1:
-            return 4;
-        default:
-            return 0;
-    }
-};
-
 // Returns a random combinator chosen so that the generated selector is valid.
-const chooseCombinator = (depth, index) => {
-    const selectors = [" ", " > "];
-    if (index > 0 && depth !== 7) {
-        selectors.push(" + ");
-        selectors.push(" ~ ");
-    }
-    return random.choice(selectors);
+const chooseCombinator = (element) => {
+    const combinators = [Combinator.DESCENDANT, Combinator.CHILD];
+    if (element.previousElementSibling)
+        combinators.push(Combinator.ADJACENT_SIBLING, Combinator.GENERAL_SIBLING);
+
+    return random.choice(combinators);
 };
 
-const randomWeighted = (options, probs) => {
-    const randNum = random.float();
-    let accumProb = 0;
-    for (let i = 0; i < probs.length; i++) {
-        accumProb += probs[i];
+// Returns a random option from the given options array, weighted by the corresponding probabilities in the probs array.
+const randomWeighted = (options, probabilities) => {
+    const randomNumber = random.float();
+    let accumulatedProbability = 0;
+    for (let i = 0; i < probabilities.length; i++) {
+        accumulatedProbability += probabilities[i];
         // prettier-ignore
-        if (randNum <= accumProb)
+        if (randomNumber <= accumulatedProbability)
             return options[i];
     }
     return options[options.length - 1];
 };
 
-const buildMatchingSelector = (depth, index, oldCombinator, selLen, maxLen) => {
-    // prettier-ignore
-    if (selLen >= maxLen)
-        return "";
-
-    const getSelector = randomWeighted([getClassname, getType, () => "*"], [0.6, 0.3, 0.1]);
-    const selector = getSelector(depth, index);
-    const combinator = chooseCombinator(depth, index);
+const buildSelectors = (element, depth, oldCombinator, selectorLength, maxSelectorLength, isMatching) => {
+    if ((!isMatching && !depth) || !element || selectorLength >= maxSelectorLength) {
```

Comment:
if `!depth` then it doesn't matter if we are on the matching case or not, right? we just stop iterating.

Reply:
nvm, I get the difference